packages/appium/lib/cli/extension-command.js

```diff
@@ -307,27 +313,31 @@ export default class ExtensionCommand {
     const {appium, name, version} = pkgJsonData;
     this.validateExtensionFields(appium);
 
-    return {...appium, pkgName: name, version};
+    // we need to omit `driverName` and `pluginName`. `_.omit()` does the job, but it isn't really
+    // type-safe.  so we're fudging it here.
+    delete appium[/** @type {string} */(`${this.type}Name`)];
```

Comment:
can ts not detect that `${this.type}Name` is a string?

Reply:
it can now, apparently.

Reply:
pretty sure there was a bug with string template types

Reply:
actually, nevermind. no, it doesn't work. no, I don't know why.